File: modules/wem-api/src/main/java/com/enonic/wem/api/aggregation/DateRangeBucket.java
package com.enonic.wem.api.aggregation;

import java.time.Instant;

public class DateRangeBucket
    extends Bucket
{
    private Instant from;

    private Instant to;


    public DateRangeBucket( final String key, final long docCount )
    {
        super( key, docCount );
    }

    public Instant getFrom()
    {
        return from;
    }

    public void setFrom( final Instant from )
    {
        this.from = from;
    }

    public Instant getTo()
    {
        return to;
    }

    public void setTo( final Instant to )
    {
        this.to = to;
    }
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/aggregation/DateRangeBucket.ts
module api.aggregation {

    export class DateRangeBucket extends Bucket {

        private from: Date;
        private to: Date;

        constructor(key: string, docCount: number) {
            super(key, docCount);
        }

        public getFrom(): Date {
            return this.from;
        }

        public getTo(): Date {
            return this.to;
        }

        public static fromDateRangeJson(json: api.aggregation.DateRangeBucketJson): DateRangeBucket {

            var dateRangeBucket: api.aggregation.DateRangeBucket = new api.aggregation.DateRangeBucket(json.key, json.docCount);
            dateRangeBucket.from = json.from;
            dateRangeBucket.to = json.to;

            return dateRangeBucket;
        }

    }

}